### PR TITLE
[New Rule] Linux Restricted Shell Breakout via the vi command

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
https://github.com/elastic/detection-rules/issues/1808

## Summary
vi, a common IOC, the Unix binary can be abused to breakout out of restricted shells or environments by spawning an interactive system shell. This activity is not standard use with this binary for a user or system administrator. It indicates a potentially malicious actor attempting to improve the capabilities or stability of their access.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/linux/defense_evasion_vi_binary.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+rule_id = "89583d1b-3c2e-4606-8b74-0a9fd2248e88"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Defense Evasion", "GTFOBins"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+sequence by host.id,process.pid with maxspan=1m
+[process where process.name == "vi" and process.args : "-c" and process.args : (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh") ]
+[process where process.parent.name == "vi" and process.name : ("bash", "sh")]
```
Comment (DefSecSentinel) [inline rules/linux/defense_evasion_vi_binary.toml:29]:
```suggestion
sequence by host.id,process.pid with maxspan=1m
[process where process.name == "vi" and process.args : "-c" and process.args : (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
[process where process.parent.name == "vi" and process.name : ("bash", "sh")]
```

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/linux/defense_evasion_vi_binary.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+[metadata]
+creation_date = "2022/03/03"
+maturity = "production"
+updated_date = "2022/03/03"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies Linux binary find abuse to break out from restricted environments by spawning an interactive system shell.
+This activity is not standard use with this binary for a user or system administrator. It indicates a potentially
```
Comment (brokensound77) [inline rules/linux/defense_evasion_vi_binary.toml:10]:
```suggestion
This activity is not standard use with this binary for a user or system administrator and could potentially indicate
```

File: rules/linux/defense_evasion_vi_binary.toml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+references = ["https://gtfobins.github.io/gtfobins/vi/"]
+risk_score = 47
+rule_id = "89583d1b-3c2e-4606-8b74-0a9fd2248e88"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Defense Evasion", "GTFOBins"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+sequence by host.id,process.pid with maxspan=1m
```
Comment (brokensound77) [inline rules/linux/defense_evasion_vi_binary.toml:27]: can you verify that when doing this by `process.entity_id` (vs `host.id,process.pid`) that it is not working as expected? They match in the data provided (and we should use that when possible)

File: rules/linux/defense_evasion_vi_binary.toml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+[process where process.name == "vi" and process.args : "-c" and process.args : (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
+[process where process.parent.name == "vi" and process.name : ("bash", "sh")]
+'''
+
+
+[[rule.threat]]
```
Comment (Samirbous) [inline rules/linux/defense_evasion_vi_binary.toml:33]: maybe u can add https://attack.mitre.org/techniques/T1059/004/

**Review by Samirbous — APPROVED** [review]

File: rules/linux/defense_evasion_vi_binary.toml (hunk 4 above)
Comment (shashank-elastic) [reply rules/linux/defense_evasion_vi_binary.toml:33]: added/

File: rules/linux/defense_evasion_vi_binary.toml (hunk 3 above)
Comment (DefSecSentinel) [reply rules/linux/defense_evasion_vi_binary.toml:27]: process.entity_id currently does not work due to the reasons provided in the archived convo I sent you but this should be fixed in 8.2 with the endpoint POSIX process event data changes

Comment (DefSecSentinel) [reply rules/linux/defense_evasion_vi_binary.toml:27]: sequence by host.id and process.pid works perfectly fine right now and when the POSIX changes get introduced we will go back and test with process.entity_id and change to that

File: rules/linux/defense_evasion_vi_binary.toml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1059"
+name = "Command and Scripting Interpreter"
+reference = "https://attack.mitre.org/techniques/T1059/"
+[[rule.threat.technique.subtechnique]]
+id = "T1059.004"
+name = "Command and Scripting Interpreter"
+reference = "https://attack.mitre.org/techniques/T1059/004/"
```
Comment (DefSecSentinel) [inline rules/linux/defense_evasion_vi_binary.toml:59]:
```suggestion
[[rule.threat.technique.subtechnique]]
id = "T1059.004"
name = "Unix Shell"
reference = "https://attack.mitre.org/techniques/T1059/004/"
```

**Pull request merged**